File: src/main/java/br/com/leonardoferreira/jirareport/domain/embedded/Changelog.java
package br.com.leonardoferreira.jirareport.domain.embedded;

import java.io.Serializable;
import java.time.LocalDateTime;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Changelog implements Serializable {
    private static final long serialVersionUID = 6432821107545516780L;

    private LocalDateTime created;

    private String from;

    private String to;

    private Long leadTime;

    private LocalDateTime endDate;
}

File: src/main/kotlin/br/com/jiratorio/domain/entity/embedded/Changelog.kt
package br.com.jiratorio.domain.entity.embedded

import br.com.jiratorio.extension.time.LocalDateProgression
import br.com.jiratorio.extension.time.rangeTo
import java.io.Serializable
import java.time.LocalDateTime

data class Changelog(

    val from: String? = null,

    val to: String? = null,

    val created: LocalDateTime,

    var endDate: LocalDateTime = created,

    var leadTime: Long = 0

) : Serializable {
    companion object {
        private const val serialVersionUID = 6432821107545516780L
    }

    fun dateRange(): LocalDateProgression {
        return created.toLocalDate()..endDate.toLocalDate()
    }
}
